src/components/GameBoard.tsx

```diff
@@ -1,42 +1,39 @@
 import * as React from 'react';
 
-interface IBoardProps {
-  board: Board;
-}
-
-class GameBoard extends React.Component<IBoardProps> {
-  public static defaultProps = {
-    board: Array(3)
-      .fill(undefined)
-      .map(() => Array(3).fill(undefined)),
+const GameBoard: React.SFC<IBoardProps> = props => {
+  const { data } = props;
+  const handleClick = (row: number, col: number): any => {
+    const coords = { row, col };
+    props.handlePlayerInput(coords);
   };
 
-  public render() {
-    const { board } = this.props;
-    return (
-      <React.Fragment>
-        {board ? (
-          <div id='board'>
-            <table id='game-table'>
-              <tbody>
-                {board.map((row: Row, rowIdx: number) => (
-                  <tr key={rowIdx} className='row'>
-                    {row.map((square: Square, squareIdx: number) => (
-                      <td key={squareIdx} className='square'>
-                        {square || '•'}
-                      </td>
-                    ))}
-                  </tr>
-                ))}
-              </tbody>
-            </table>
-          </div>
-        ) : (
-          <h1>NO_BOARD</h1>
-        )}
-      </React.Fragment>
-    );
-  }
-}
+  return (
+    <React.Fragment>
+      {data ? (
+        <div id='board'>
+          <table id='game-table'>
+            <tbody>
+              {data.map((row: Row, rowIdx: number) => (
```

Comment:
Now that Typescript knows what `data` is, you shouldn't have to type these arguments, they will be inferred correctly.